### PR TITLE
Process reconnect when connectionstatechange callback not called

### PR DESCRIPTION
**Pull request by cnderrauber**
in case of ice restart, there is a chance that neither iceconnectionstatechange or connectionstatechange will be called, there is no ice conn or pc conn state transit (sticky on connected). in this case, detect peerconnection state is still connected after 2 second.

### CODE REVIEW

File: src/room/RTCEngine.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ import { sleep } from './utils';
 const lossyDataChannel = '_lossy';
 const reliableDataChannel = '_reliable';
 const maxReconnectRetries = 5;
+const maxReconnectingTimeout = 2 * 1000;
```
Comment (davidzhao) [inline src/room/RTCEngine.ts:18]: do you think a minimum wait for 1s is sufficient to allow connected to flip to connecting?

```suggestion
const minReconnectWait = 1 * 1000;
```

File: src/room/RTCEngine.ts [hunk 2]
```diff
@@ -39,6 +40,8 @@ export default class RTCEngine extends EventEmitter {
 
   private subscriberPrimary: boolean = false;
 
+  private pcConnectionState: RTCPeerConnectionState = 'new';
```
Comment (davidzhao) [inline src/room/RTCEngine.ts:43]: can we not store this var, and check primaryPC.connectionState?

File: src/room/RTCEngine.ts (hunk 1 above)
Comment (cnderrauber) [reply src/room/RTCEngine.ts:18]: I'm not sure, 2s just a initial value. need more observe

File: src/room/RTCEngine.ts [hunk 3]
```diff
@@ -372,13 +376,20 @@ export default class RTCEngine extends EventEmitter {
 
     const startTime = (new Date()).getTime();
 
-    while ((new Date()).getTime() - startTime < maxICEConnectTimeout) {
+    let now = startTime;
+    while (now - startTime < maxICEConnectTimeout) {
+      // if there is no connectionstatechange callback fired
+      // check connectionstate after maxReconnectingTimeout
```
Comment (davidzhao) [inline src/room/RTCEngine.ts:382]: nit: update comment too

```suggestion
      // check connectionstate after minReconnectWait
```

**Review by davidzhao — APPROVED** [review]

**Pull request merged**